src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+]])
+
+M['iproto.advertise.peer.params'] = M['<uri>.params']
+
+M['iproto.advertise.peer.params.ssl_ca_file'] = M['<uri>.params.ssl_ca_file']
+
+M['iproto.advertise.peer.params.ssl_cert_file'] =
+    M['<uri>.params.ssl_cert_file']
+
+M['iproto.advertise.peer.params.ssl_ciphers'] = M['<uri>.params.ssl_ciphers']
+
+M['iproto.advertise.peer.params.ssl_key_file'] = M['<uri>.params.ssl_key_file']
+
+M['iproto.advertise.peer.params.ssl_password'] = M['<uri>.params.ssl_password']
+
+M['iproto.advertise.peer.params.ssl_password_file'] =
+    M['<uri>.params.ssl_password_file']
+
+M['iproto.advertise.peer.params.transport'] = M['<uri>.params.transport']
+
+M['iproto.advertise.peer.password'] = format_text([[
+    (Optional) A password for the specified user. If a login is specified but
+    a password is missing, it is taken from the user's credentials.
+]])
+
+M['iproto.advertise.peer.uri'] = format_text([[
+    (Optional) A URI used to advertise the current instance. By default,
+    the URI defined in iproto.listen is used to advertise the current instance.
+]])
+
+M['iproto.advertise.sharding'] = format_text([[
+    Settings used to advertise the current instance to a router and
+    rebalancer. The format of these settings is described in
+    `iproto.advertise.<peer_or_sharding>.*`.
+]])
+
+M['iproto.advertise.sharding.login'] = format_text([[
+    (Optional) A username used to connect to the current instance.
+    If a username is not set, the guest user is used.
+]])
+
+M['iproto.advertise.sharding.params'] = M['<uri>.params']
+
+M['iproto.advertise.sharding.params.ssl_ca_file'] =
+    M['<uri>.params.ssl_ca_file']
+
+M['iproto.advertise.sharding.params.ssl_cert_file'] =
+    M['<uri>.params.ssl_cert_file']
+
+M['iproto.advertise.sharding.params.ssl_ciphers'] =
+    M['<uri>.params.ssl_ciphers']
+
+M['iproto.advertise.sharding.params.ssl_key_file'] =
+    M['<uri>.params.ssl_key_file']
+
+M['iproto.advertise.sharding.params.ssl_password'] =
+    M['<uri>.params.ssl_password']
+
+M['iproto.advertise.sharding.params.ssl_password_file'] =
+    M['<uri>.params.ssl_password_file']
+
+M['iproto.advertise.sharding.params.transport'] = M['<uri>.params.transport']
+
+M['iproto.advertise.sharding.password'] = format_text([[
+    (Optional) A password for the specified user. If a login is specified but
+    a password is missing, it is taken from the user's credentials.
+]])
+
+M['iproto.advertise.sharding.uri'] = format_text([[
+    (Optional) A URI used to advertise the current instance. By default,
+    the URI defined in iproto.listen is used to advertise the current instance.
+]])
+
+-- }}} iproto.advertise configuration
+
+M['iproto.listen'] = format_text([[
+    An array of URIs used to listen for incoming requests. If required,
+    you can enable SSL for specific URIs by providing additional parameters
+    (`<uri>.params.*`).
+
+    Note that a URI value can't contain parameters, a login, or a password.
```

Comment:
If 'parameters' here are SSL/TLS ones, `iproto.listen` contains them.

Reply:
Ah, got it. It is about `login@host:port`, `login:password@host:port`, `host:port?transport=ssl&ssl_cert_file=<...>&ssl_key_file=<...>` and so on.

I would say something like the following:

> The `iproto.listen.*.uri` string can't contain a login or a password, it has no sense for a listening socket.
>
> The query-parameter form of setting SSL options is forbidden in the URI string.
> Use the `iproto.listen.*.params` for them.

And place it to the `iproto.listen.*.uri` description rather than to `iproto.listen`.

Reply:
Done.